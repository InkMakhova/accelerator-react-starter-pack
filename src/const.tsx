export enum PageTitle {
  Catalog = 'Guitar-shop',
  Cart = 'Корзина — Guitar-shop',
}

export enum AppRoute {
  Root = '/',
  Catalog = '/catalog',
  Page = '/page_',
  Cart = '/cart',
  WhereToBuy = '/where-to-buy',
  AboutUs = '/about-us',
  NotFoundPage = '/page-not-found',
  Guitars = '/guitars/',
}

export enum APIRoute {
  Guitars = '/Guitars',
}

export enum QueryParam {
  SortParam = '_sort',
  OrderParam = '_order',
  PriceMinParam = 'price_gte',
  PriceMaxParam = 'price_lte',
  TypeParam = 'type[]',
  StringCountParam = 'stringCount[]',
  StartParam = '_start',
  LimitParam = '_limit',
  SearchParam = 'name_like',
}

export const ITEMS_PER_PAGE = 9;

export const PAGINATION_STEP = 3;

export enum Sort {
  Price = 'price',
  Rating = 'rating',
}

export enum Order {
  Asc = 'Asc',
  Desc = 'Desc',
}

export enum Type {
  Acoustic= 'acoustic',
  Electric = 'electric',
  Ukulele = 'ukulele',
}

export enum StringCount {
  Four = 4,
  Six = 6,
  Seven = 7,
  Twelve = 12,
}

export const PRICE_MIN = 0;
export const PRICE_MAX = 0;
