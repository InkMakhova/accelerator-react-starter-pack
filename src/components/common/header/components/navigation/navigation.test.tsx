import React from 'react';
import { render } from '@testing-library/react';
import { BrowserRouter as Router } from 'react-router-dom';
import Navigation from './navigation';
import {NavigationSection} from '../../../../../const';

it('should render Navigation correctly', () => {
  const {container} = render(
    <Router>
      <Navigation currentNavigationSection={NavigationSection.Catalog}/>
    </Router>);
  expect(container).toMatchSnapshot();
});
