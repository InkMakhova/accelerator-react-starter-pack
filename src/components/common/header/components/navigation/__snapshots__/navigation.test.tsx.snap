// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render Navigation correctly 1`] = `
<div>
  <nav
    class="main-nav"
  >
    <ul
      class="main-nav__list"
    >
      <li>
        <a
          class="link main-nav__link
              link--current"
          href="/"
        >
          Каталог
        </a>
      </li>
      <li>
        <a
          class="link main-nav__link
              "
          href="/where-to-buy"
        >
          Где купить?
        </a>
      </li>
      <li>
        <a
          class="link main-nav__link
              "
          href="/about-us"
        >
          О компании
        </a>
      </li>
    </ul>
  </nav>
</div>
`;
